diff --git a/src/platform/tls_openssl.c b/src/platform/tls_openssl.c
index 61c42fe..b4a8b0d 100644
--- a/src/platform/tls_openssl.c
+++ b/src/platform/tls_openssl.c
@@ -995,7 +995,9 @@ CxPlatTlsCertificateVerifyCallback(
     if ((TlsContext->SecConfig->Flags & QUIC_CREDENTIAL_FLAG_CLIENT ||
         IsDeferredValidationOrClientAuth) &&
         !(TlsContext->SecConfig->Flags & QUIC_CREDENTIAL_FLAG_NO_CERTIFICATE_VALIDATION)) {
-        if (!(TlsContext->SecConfig->Flags & QUIC_CREDENTIAL_FLAG_USE_TLS_BUILTIN_CERTIFICATE_VALIDATION)) {
+        if (!(TlsContext->SecConfig->Flags &
+            (QUIC_CREDENTIAL_FLAG_USE_TLS_BUILTIN_CERTIFICATE_VALIDATION |
+             QUIC_CREDENTIAL_FLAG_SET_CA_CERTIFICATE_FILE))) {
             if (Cert == NULL) {
                 QuicTraceEvent(
                     TlsError,
diff --git a/src/platform/tls_quictls.c b/src/platform/tls_quictls.c
index 668e925..24b710d 100644
--- a/src/platform/tls_quictls.c
+++ b/src/platform/tls_quictls.c
@@ -244,7 +244,9 @@ CxPlatTlsCertificateVerifyCallback(
     if ((TlsContext->SecConfig->Flags & QUIC_CREDENTIAL_FLAG_CLIENT ||
         IsDeferredValidationOrClientAuth) &&
         !(TlsContext->SecConfig->Flags & QUIC_CREDENTIAL_FLAG_NO_CERTIFICATE_VALIDATION)) {
-        if (!(TlsContext->SecConfig->Flags & QUIC_CREDENTIAL_FLAG_USE_TLS_BUILTIN_CERTIFICATE_VALIDATION)) {
+        if (!(TlsContext->SecConfig->Flags &
+            (QUIC_CREDENTIAL_FLAG_USE_TLS_BUILTIN_CERTIFICATE_VALIDATION |
+             QUIC_CREDENTIAL_FLAG_SET_CA_CERTIFICATE_FILE))) {
             if (Cert == NULL) {
                 QuicTraceEvent(
                     TlsError,
